### PR TITLE
Test and build simultaneously, smarter nightly job

### PR DESCRIPTION
**Pull request by dconnolly**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -10,27 +10,29 @@ cache:
...
   - "test and build"
 
 jobs:
+  fast_finish: true
   allow_failures:
     - rust: nightly-2019-06-30
   include:
-    - stage: fetch
-      name: "cargo fetch --verbose"
+    - name: "cargo fetch --verbose"
```
Comment (hdevalence) [inline .travis.yml]: Does this still have an effect if it runs simultaneously with the other CI stages, or will each `cargo test/build/etc` do fetches on its own?

Comment (dconnolly) [reply .travis.yml]: The `fetch` stage always goes first (as specified by the `stages` section above), so the jobs in the `test and build` stage that comes next will be pre-fetched; if they do run `fetch` as part of test and build they will be doing less work I think.

Comment (dconnolly) [reply .travis.yml]: eg:

image

**Comment by dconnolly** [comment]
Hm.
image

**Comment by dconnolly** [comment]
> Hm.
> image

Ah. I think when we run `fetch` in the first stage, and then `test` and `build` in the next stage in parallel, and because they inherit the same pre`fetch`d workspace, `build` blocks on the `.cargo/` registry locks established by `test`. Boo.

**Comment by dconnolly** [comment]
> > Hm.
> > image
> 
> Ah. I think when we run `fetch` in the first stage, and then `test` and `build` in the next stage in parallel, and because they inherit the same pre`fetch`d workspace, `build` blocks on the `.cargo/` registry locks established by `test`. Boo.

OK confirmed, now everything just runs in parallel, no `fetch` job first. Merging.

File: .travis.yml (hunk 1 above)
Comment (hdevalence) [reply .travis.yml]: Sweet!

**Pull request merged**